crates/rooch-types/src/address.rs

```diff
@@ -821,7 +814,32 @@ impl TryFrom<MultiChainAddress> for BitcoinAddress {
 // Ref: https://github.com/nostr-protocol/nips/blob/master/19.md
 /// Nostr public key type
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub struct NostrPublicKey(pub XOnlyPublicKey);
+pub struct NostrPublicKey(XOnlyPublicKey);
+
+impl NostrPublicKey {
+    pub fn new(x_only_pk: XOnlyPublicKey) -> Self {
+        Self(x_only_pk)
+    }
+
+    /// Convert from the Nostr XOnlyPublicKey to Bitcoin Taproot address. BIP-086.
+    pub fn to_bitcoin_address(&self, network: u8) -> Result<BitcoinAddress, anyhow::Error> {
+        // get the network
+        let network = network::Network::try_from(network)?;
+        // change use of XOnlyPublicKey from nostr to bitcoin lib
+        let internal_key = bitcoin::XOnlyPublicKey::from_slice(&self.0.serialize())?;
+        // new verification crypto
+        let secp = Secp256k1::verification_only();
+        // new bitcoin taproot address
+        let address = Address::p2tr(
+            &secp,
+            internal_key,
+            None,
+            bitcoin::network::Network::from(network),
+        );
+        // give it to rooch bitcoin struct
+        Ok(BitcoinAddress::new(address.to_string().as_bytes().to_vec()))
```

Comment:
BitcoinAddress::from(bitcoin::Address), not address.to_string()

Reply:
Oh Okay.